odoo/fields.py

```diff
@@ -1783,10 +1785,25 @@ def read(self, records):
         for record in records:
             cache.set(record, self, data.get(record.id, False))
 
+    def _check_svg(self, record, value):
+        """Detect if the binary content is an SVG for restricting its upload
+        only to system users.
+        """
+        if not value:
+            return
+        if value[:1] == b'P':  # Fast detection of first 6 bits of '<' (0x3C)
+            decoded_value = base64.b64decode(value)
+            # Full mimetype detection
+            if (guess_mimetype(decoded_value) == 'image/svg+xml' and
+                    not record.user_has_groups('base.group_system')):
+                raise UserError("Only admins can upload SVG files.")
+
     def create(self, record_values):
         assert self.attachment
         if not record_values:
             return
+        for record, value in record_values:
+            self._check_svg(record, value)
```

Comment:
I think this check in `create()` and `write()` is redundant with the neutering of ir.attachment, and should be done in convert_to_column instead.